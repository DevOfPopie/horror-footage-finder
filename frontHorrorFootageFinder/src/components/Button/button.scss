@use '../../assets/styles/vars';

.button-ui {
    width: fit-content;
    font-size: 1.3rem;
    background: white;
    border: 3px solid transparent;
    font-family: inherit;
    box-shadow: 4px 4px white;
    cursor: pointer;
    box-sizing: border-box;
    font-weight: bold;
    text-transform: uppercase;

    a {
        color: vars.$black-primary;
        display: inline-block;
        padding: 1rem;
    }

    &__content {
        padding: 1rem;
    }

    &__launch {
        font-size: vars.$h2-font-size;
    }

    &__selected, &:hover {
        background: vars.$black-primary;
        border: 3px solid white;
        color: white;

        a {
            color: white;
        }
    }
}