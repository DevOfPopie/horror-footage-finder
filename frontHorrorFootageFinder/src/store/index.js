import { createStore, applyMiddleware } from 'redux';
import thunk from 'redux-thunk';
import { composeWithDevTools } from 'redux-devtools-extension';
import reducer from '../reducers';
import dataHorror from '../middlewares/dataHorror';
import watchList from '../middlewares/watchList';

const store = createStore(
  reducer,
  composeWithDevTools(
    applyMiddleware(dataHorror, watchList, thunk),
  ),
);

export default store;
