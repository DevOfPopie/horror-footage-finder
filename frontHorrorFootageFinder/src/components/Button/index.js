import React from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import { NavLink } from 'react-router-dom';

import './button.scss';

export function Button({
  textContent, onClick, to, className, type,
}) {
  return (
    <button
      className={`button-ui ${className}`}
      onClick={onClick}
      // eslint-disable-next-line react/button-has-type
      type={type}
    >
      {to ? <NavLink to={to}>{textContent}</NavLink> : textContent }
    </button>
  );
}

Button.propTypes = {
  textContent: PropTypes.string.isRequired,
  onClick: PropTypes.func,
  to: PropTypes.string,
  className: PropTypes.string,
  type: PropTypes.string,
};

Button.defaultProps = {
  onClick: () => {},
  to: null,
  className: '',
  type: 'button',
};

const mapStateToProps = () => ({
});

const mapDispatchToProps = () => ({

});

export default connect(mapStateToProps, mapDispatchToProps)(Button);
