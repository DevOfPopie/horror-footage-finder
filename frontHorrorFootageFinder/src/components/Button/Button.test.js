import React from 'react';
import ReactTestRenderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import { render, screen } from '@testing-library/react';

import store from '../../store';
import Button from './index';

describe('Button testing', () => {
  // Renders readable textContent
  it('renders message via textContent', () => {
    render(
      <Provider store={store}>
        <Button textContent="Animations activées" />
      </Provider>,
    );
    expect(screen.getByText('Animations activées')).toBeInTheDocument();
  });

  // v1 snapshot of Button
  it('renders as the v1 version of Button comoponent with className, onClick, selected, type & value', () => {
    const component = ReactTestRenderer.create(
      <Provider store={store}>
        <Button textContent="Animations activées" />
      </Provider>,
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
