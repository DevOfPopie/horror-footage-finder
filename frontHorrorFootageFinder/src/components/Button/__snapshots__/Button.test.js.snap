// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button testing renders as the v1 version of Button comoponent with className, onClick, selected, type & value 1`] = `
<button
  className="button-ui  false button-ui__animated"
  onClick={[Function]}
  selected={false}
  type="button"
  value={null}
>
  <div
    className="button-ui__content"
  >
    Animations activ√©es
  </div>
</button>
`;
